plutus-core/plutus-ir/src/PlutusIR/Transform/Beta.hs

```diff
@@ -114,4 +125,8 @@ beta = \case
     -- See Note [Multi-beta]
     -- This maybe isn't the best annotation for this term, but it will do.
     (extractBindings -> Just (bs, t)) -> Let (termAnn t) NonRec bs (beta t)
```

Comment:
Hm. Why did I write this pass with explicit recursion rather than using the usual `transformOf` pattern? That's quite suspicious. In particular, it means that we're not actually processing the subterms that get pulled into the let-bindings here. So something like this probably won't get handled properly: `(\a . t) ((\b . u) x)`. I think it will become `let a = (\b . u) x in t` instead of `let a = (let b = x in u) in t`.

One for a follow-up PR, perhaps.

Reply:
Maybe add that as a test case anyway just to check.

Reply:
You are right. Added a test case.